object-cache.php

```diff
@@ -711,56 +720,55 @@ function __construct() {
 			}
 		}
 
-		$this->redis = new WP_Redis();
-		$this->redis->wp_object_cache = &$this;
+		$this->redis = new Redis;
 		$this->redis->connect( $redis_server['host'], $redis_server['port'], 1, NULL, 100 ); # 1s timeout, 100ms delay between reconnections
 		if ( ! empty( $redis_server['auth'] ) ) {
 			$this->redis->auth( $redis_server['auth'] );
 		}
-
-		$this->global_prefix = '';
-		if ( function_exists( 'is_multisite' ) ) {
-			$this->global_prefix = ( is_multisite() || defined( 'CUSTOM_USER_TABLE' ) && defined( 'CUSTOM_USER_META_TABLE' ) ) ? '' : $table_prefix;
+		$this->is_redis_connected = $this->_call_redis( 'isConnected' );
+		if ( ! $this->is_redis_connected ) {
+			$this->missing_redis_message = 'Alert! WP Redis object cache cannot connect to Redis server.';
 		}
-
-		/**
-		 * @todo This should be moved to the PHP4 style constructor, PHP5
-		 * already calls __destruct()
-		 */
-		register_shutdown_function( array( $this, '__destruct' ) );
+		return $this->is_redis_connected;
 	}
 
 	/**
-	 * Will save the object cache before object is completely destroyed.
+	 * Wrapper method for calls to Redis, which fails gracefully when Redis is unavailable
 	 *
-	 * Called upon object destruction, which should be when PHP ends.
-	 *
-	 * @return bool True value. Won't be used by PHP
+	 * @param string $method
+	 * @param mixed $args
+	 * @return mixed
 	 */
-	function __destruct() {
-		return true;
-	}
-}
-
-if ( class_exists( 'Redis' ) ) {
-	class WP_Redis extends Redis {
-
-	}
-} else {
-	class WP_Redis {
+	protected function _call_redis( $method ) {
```

Comment:
I'm not a huge fan of prefixing methods with `_`, but the pattern is already established with `_exists`